internal/controller/build/controller.go

```diff
@@ -306,41 +306,54 @@ func (r *Reconciler) makeBuildContext(ctx context.Context, build *choreov1.Build
 	}, nil
 }
 
-func (r *Reconciler) getDPClient(ctx context.Context, build *choreov1.Build) (client.Client, error) {
-	dataplaneRes, err := getDataplane(ctx, r.Client, build)
+func (r *Reconciler) getBPClient(ctx context.Context, build *choreov1.Build) (client.Client, error) {
+	buildplane, err := r.getDataPlaneMarkedAsBuildPlane(ctx, r.Client, build)
 	if err != nil {
 		// Return an error if dataplane retrieval fails
-		return nil, fmt.Errorf("failed to get default dataplane %s: %w", "default-dataplane", err)
+		return nil, fmt.Errorf("failed to get build plane: %w", err)
 	}
 
-	dpClient, err := dpKubernetes.GetDPClient(r.DpClientMgr, dataplaneRes)
+	dpClient, err := dpKubernetes.GetDPClient(r.DpClientMgr, buildplane)
 	if err != nil {
 		// Return an error if client creation fails
-		return nil, fmt.Errorf("failed to get DP client: %w", err)
+		return nil, fmt.Errorf("failed to get build plane client: %w", err)
 	}
 
 	return dpClient, nil
 }
 
-func getDataplane(ctx context.Context, c client.Client, build *choreov1.Build) (*choreov1.DataPlane, error) {
-	dataplaneList := &choreov1.DataPlaneList{}
+func (r *Reconciler) getDataPlaneMarkedAsBuildPlane(ctx context.Context, c client.Client, build *choreov1.Build) (*choreov1.DataPlane, error) {
+	dataplanes := &choreov1.DataPlaneList{}
 	listOpts := []client.ListOption{
 		client.InNamespace(build.GetNamespace()),
```

Comment:
can't we add the buildplane label here and get only the dataplane marked as build plane?